### PR TITLE
Feature/mex 157/pull message

### PR DESCRIPTION
**Pull request by voscob**
### Description

Please include a summary of the change and what type of change it is (new feature, bug fix, refactoring, documentation).
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Fixes # (issue)

### Quick checks:

- [ ] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [ ] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-gcp-pubsub/pulls) for the same update/change.
- [ ] I have written unit tests.
- [ ] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: config/general.go [hunk 1]
```diff
@@ -19,22 +19,23 @@ import (
 	"github.com/conduitio/conduit-connector-gcp-pubsub/models"
 )
 
-type general struct {
-	PrivateKey  string `validate:"required"`
-	ClientEmail string `validate:"required,email"`
-	ProjectID   string `validate:"required"`
+// A General represents a general configuration needed for GCP Pub/Sub.
+type General struct {
+	PrivateKey  string `json:"private_key" validate:"required"`
```
Comment (maksenius) [inline config/general.go:24]: please add short description

File: clients/pubsub.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	}
+
+	cli, err := pubsub.NewClient(ctx, cfg.ProjectID, option.WithCredentialsJSON(credential))
+	if err != nil {
+		return PubSub{}, fmt.Errorf("new pubsub client: %w", err)
+	}
+
+	pubSub := PubSub{
+		Cli:       cli,
+		MessageCh: make(chan *pubsub.Message),
```
Comment (maksenius) [inline clients/pubsub.go:51]: Why not buffered chan?

File: source/source.go [hunk 3]
```diff
@@ -43,3 +47,41 @@ func (s *Source) Configure(ctx context.Context, cfgRaw map[string]string) error
 
 	return nil
 }
+
+// Open initializes a Pub/Sub client.
+func (s *Source) Open(ctx context.Context, _ sdk.Position) error {
+	pubSub, err := clients.NewClient(ctx, s.cfg)
```
Comment (maksenius) [inline source/source.go:53]: Did we do ping there?

File: config/general.go (hunk 1 above)
Comment (voscob) [reply config/general.go]: Deal!

File: clients/pubsub.go (hunk 2 above)
Comment (voscob) [reply clients/pubsub.go]: You're right!
Done.

File: source/source.go (hunk 3 above)
Comment (voscob) [reply source/source.go:51]: No, there is no ping.
If the credentials are invalid, the system returns an error while creating a reference to a subscription:
https://github.com/conduitio-labs/conduit-connector-gcp-pubsub/blob/feature/MEX-157/pull-message/clients/pubsub.go#L62

**Review by maksenius — APPROVED** [review]

**Pull request merged**